lib/cfg.py

```diff
@@ -251,6 +253,83 @@ def cfg_version(filename):
         cfg[section] = d
     return cfg
 
+def cfg_altiumWorkspace(options):
+    """
+    ricerca del nome di tutti i progetti all'interno del file Workspace
+    """
+    risultato=True
+    path_dict = {}
+    config = ConfigParser.RawConfigParser()
+    config.read(options.ws)
+
+    for i in config.sections():
+        try:
+            temp = config.get(i, 'ProjectPath')
+            temp = temp.split('/')
+            k = []
+            p = []
+
+            if len(temp) > 1:
+                k = os.path.splitext(temp[1])[0]
+                p = os.path.join(*temp)
+            else:
+                k = os.path.splitext(temp[0])[0]
+                p = os.path.join("./", *temp)
+
+            if len(k) == 0 or len(p) == 0:
+                lib.error("Errore nel parsing del workspace: %s" % e,
+                            self.handler, terminal=self.terminal)
+                risultato =False
+
+            k = k.lower()
+            path_dict[k] = p
+
+        except ConfigParser.NoOptionError:
+            pass
+    path=''
+    ws=options.ws.split('/')
+    for i in range(0, (len(ws)-1)):
+        path=os.path.join(path, ws[i])
+
+    """
+    ricerca parametri per ogni progetto e creazione dizionario con {parametro: valore}
+    ricerca e verifica esistenza file a cui fare il mergebom e messi in una lista
+    """
+    parametri_dict={}
+    file_BOM=[]
+    if risultato:
+        nprogetti=0 
+        for k, v in path_dict.items():
+
+            prj=os.path.join(path,v)
+            f=open(prj,'r')
+            config=ConfigParser.RawConfigParser()
+            config.read(prj)
+
+            path_filemerge=os.path.join(path,  "Assembly")
+            for i in config.sections():
+                line = re.findall(r'Parameter[0-9]', i)
+
+                if line:
+                    parametro=config.get(i,'Name')
+                    val=config.get(i,'Value')
+                    parametri_dict[parametro]=val
+            pathfile=os.path.join(path_filemerge,k)
+            if options.csv_file:
+                filecsv = os.path.join(pathfile, k)+'.csv'
+                if os.path.exists(filecsv):
+                    file_BOM.append(filecsv)
+            else:
+                filexlsx = os.path.join(pathfile, k) +'.xlsx'
+                if os.path.exists(filexlsx):
+                    file_BOM.append(filexlsx)
+                
+            nprogetti+=1
+    file=(file_BOM, parametri_dict)
```

Comment:
Il valore di ritorno, non è quello che avevamo detto.. perchè il workspace può contenere più progetti quindi tu avrai più set di dati per ogni progetto, e quindi dovrai tornare una lista, dove per ogni elemento ci sono i dati di ogni singolo progetto:

```
ret = [
    ([file1.csv, file2.csv], parametri_prj1),
    ([file1.csv, file2.csv], parametri_prj2),
    ([file1.csv, file2.csv], parametri_prj3),
    ([file1.csv, file2.csv], parametri_prj3),
    ...
]
```
Quindi questo pezzo di codice va corretto per gestire questa struttura dati.

Altra cosa, che qui siamo in python e il contatore non ti serve, perchè i tipi sono dinamici e ci si può sempre iterare sopra..